src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/Ordering/OrderingHelper.cs

```diff
@@ -279,9 +259,20 @@ private static bool IsValidDisplayOrder(int displayOrder)
         /// Gets a collection a project tree's children. 
         /// The children will only have a valid display order, and the collection will be in order by their display order.
         /// </summary>
-        private static ImmutableArray<IProjectTree> GetChildren(IProjectTree projectTree)
+        private static ImmutableArray<IProjectTree> GetChildren(Project project, IProjectTree projectTree)
         {
-            return projectTree.Children.Where(x => HasValidDisplayOrder(x)).OrderBy(x => GetDisplayOrder(x)).ToImmutableArray();
+            return projectTree.Children.Where(x =>
+            {
+                var hasValidDisplayOrder = HasValidDisplayOrder(x);
+                if (!hasValidDisplayOrder)
+                {
+                    return false;
+                }
+
+                // Make sure we have a valid backing xml element that is not part of a .props or .targets files.
+                return GetItemElements(project, x, ImmutableArray<string>.Empty).Any();
```

Comment:
This pattern (seen a couple of places) coupled with the `ToImmutableArray` in `GetItemElements`, forces you to walk the full tree just to see if it's non empty.  Consider having two methods `GetItemElementsDeferred` and `GetItemElements`, that does the `ToImmutableArray` - then places that use `Any` can use the deferred one and stop at the first `true` value if that's all it's used for.

Reply:
That's what I was thinking before. I can revisit this if we think that is wroth it for perf (I think so)